datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+            # There is a metadata.csv file but no entry in the manifest.json
+            self.fail('  metadata.csv exists but not defined in the manifest.json of {}'.format(check_name))
+        elif metadata_in_manifest and not metadata_file_exists:
+            # There is an entry in the manifest.json file but the referenced csv file does not exist.
+            self.fail('  metrics_metadata in manifest.json references a non-existing file: {}.'.format(metadata_file))
+
+
+class MetricToCheckValidator(BaseManifestValidator):
+    METRIC_TO_CHECK_EXCLUDE_LIST = {
+        'openstack.controller',  # "Artificial" metric, shouldn't be listed in metadata file.
+        'riakcs.bucket_list_pool.workers',  # RiakCS 2.1 metric, but metadata.csv lists RiakCS 2.0 metrics only.
+    }
+    METADATA_PATH = {"v1": "/assets/metrics_metadata", "v2": "/assets/integration/metrics/metadata_path"}
+    METRIC_PATH = {"v1": "/metric_to_check", "v2": "/assets/integration/metrics/check"}
+    PRICING_PATH = {"v1": "/pricing", "v2": "/pricing"}
+
+    def validate(self, check_name, decoded, _):
+        if not self.should_validate() or check_name == 'snmp' or check_name == 'moogsoft':
+            return
+
+        metadata_path = self.METADATA_PATH[self.version]
+        metadata_in_manifest = decoded.get_path(metadata_path)
+
+        # metric_to_check
+        metric_path = self.METRIC_PATH[self.version]
+        metric_to_check = decoded.get_path(metric_path)
+
+        pricing = decoded.get_path("/pricing") or []
+
+        if metric_to_check:
+            metrics_to_check = metric_to_check if isinstance(metric_to_check, list) else [metric_to_check]
+            if any(p.get('metric') in metrics_to_check for p in pricing):
+                return
+            for metric in metrics_to_check:
+                metric_integration_check_name = check_name
+                # snmp vendor specific integrations define metric_to_check
+                # with metrics from `snmp` integration
+                if check_name.startswith('snmp_') and not metadata_in_manifest:
+                    metric_integration_check_name = 'snmp'
+                if (
+                    not is_metric_in_metadata_file(metric, metric_integration_check_name)
+                    and metric not in self.METRIC_TO_CHECK_EXCLUDE_LIST
+                ):
+                    self.fail(f'  metric_to_check not in metadata.csv: {metric!r}')
+
+        elif metadata_in_manifest:
+            # if we have a metadata.csv file but no `metric_to_check` raise an error
+            metadata_file = get_metadata_file(check_name)
+            if os.path.isfile(metadata_file):
+                for _, row in read_metadata_rows(metadata_file):
+                    # there are cases of metadata.csv files with just a header but no metrics
+                    if row:
+                        self.fail('  metric_to_check not included in manifest.json')
+
+
+class IsPublicValidator(BaseManifestValidator):
```

Comment:
Can we name these after the new v2 versions? We tried to make the field names more true to what they're doing, and naming the validations after that may be more clear for people